dts/arm/nuvoton/npcx/npcx7/npcx7-pinctrl.dtsi

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2022 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&pinctrl {
+	/* Nodes for peripheral device's default pin-muxing and pad properties */
+	/* Host peripheral interfaces */
+	lpc_espi_default: periph-lpc-espi {
+		group0 {
+			pinmux = <&alt1_no_lpc_espi>;
+		};
```

Comment:
-1 on providing pinmux defaults at SoC level, all these nodes should be part of `board-pinctrl.dtsi` file to be consistent with pinctrl guidelines and other implementations. What can be provided, of course, are pre-defined valid configurations (in the form of nodes or definitions). Since the `pinmux` is a DT node, you can add there as many properties as you want (see for example the case in some NXP devices), so the `periph-pupd` can likely be encoded there as well.

Another question: I can see groups being used here, but I have yet to see a node with more than a group... so I wonder why use groups here?

Reply:
> -1 on providing pinmux defaults at SoC level, all these nodes should be part of `board-pinctrl.dtsi` file to be consistent with pinctrl guidelines and other implementations. What can be provided, of course, are pre-defined valid configurations (in the form of nodes or definitions). Since the `pinmux` is a DT node, you can add there as many properties as you want (see for example the case in some NXP devices), so the `periph-pupd` can likely be encoded there as well.

I think npcx is not the only one who puts the default pinctrl nodes in dts folder. (Please refer to the microchip's [solution](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/microchip/mec172x/mec172xnsz-pinctrl.dtsi).) So far, all three ec vendors (including [ITE](https://github.com/zephyrproject-rtos/zephyr/pull/44786/files#diff-a108abe857e7ce219c00e3862c40c50bc6aca1b65d5fe21f53f88fc93f6494b2)) of chromebook projects provide a similar pinctrl mechanism ([Provide the default pinctrl nodes](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/microchip/mec172x/mec172xnsz-pinctrl.dtsi#L197) and [overwrite them if needed in board DT file](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mec172xevb_assy6906/mec172xevb_assy6906.dts#L149).) In ec's application, plenty of pinctrl nodes are needed and most of them are the same in different board-pinctrl.dtsi files. If users are not familiar with the target SoC, it might be easy to put the wrong configurations in the properties of pinctrl nodes. From SoC vendor's point of view, I have no comment about the location of pinctrl nodes. (Although I got a little confused that the microchip's solution seems doesn't follow the principle.) Since it impacts the board DT drivers which are maintained by our users, @keith-zephyr, might I know your comment? 

> Another question: I can see groups being used here, but I have yet to see a node with more than a group... so I wonder why use groups here?

Most of them are in the next [PR](https://github.com/MulinChao/zephyr/commits/npcx_psl_gpio). It includes:
1. IO pads initialization. The example pinctrl node is [here](https://github.com/MulinChao/zephyr/commit/9d6a6d31435aa90edaf4cdb156c659b8812d365a#diff-4d93322284ba53780ba3d55b549effcd96b0736265b9805deee1d038121f5884R96).
2. Detection configurations of PSL input pads. The example pinctrl node is [here](https://github.com/MulinChao/zephyr/blob/npcx_test/boards/arm/npcx7m6fb_evb/npcx7m6fb_evb.dts#L84).
3. Keyboard driver. It hasn't been implemented yet. But we plan to use two groups to configure keyboard output and input pads seperately.

Reply:
> > -1 on providing pinmux defaults at SoC level, all these nodes should be part of `board-pinctrl.dtsi` file to be consistent with pinctrl guidelines and other implementations. What can be provided, of course, are pre-defined valid configurations (in the form of nodes or definitions). Since the `pinmux` is a DT node, you can add there as many properties as you want (see for example the case in some NXP devices), so the `periph-pupd` can likely be encoded there as well.
> 
> I think npcx is not the only one who puts the default pinctrl nodes in dts folder. (Please refer to the microchip's [solution](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/microchip/mec172x/mec172xnsz-pinctrl.dtsi?rgh-link-date=2022-04-28T07%3A45%3A12Z).) So far, all three ec vendors (including [ITE](https://github.com/zephyrproject-rtos/zephyr/pull/44786/files#diff-a108abe857e7ce219c00e3862c40c50bc6aca1b65d5fe21f53f88fc93f6494b2)) of chromebook projects provide a similar pinctrl mechanism ([Provide the default pinctrl nodes](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/microchip/mec172x/mec172xnsz-pinctrl.dtsi?rgh-link-date=2022-04-28T07%3A45%3A12Z#L197) and [overwrite them if needed in board DT file](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/mec172xevb_assy6906/mec172xevb_assy6906.dts?rgh-link-date=2022-04-28T07%3A45%3A12Z#L149).)

Microchip case is different, they provide pre-defined nodes with all valid combinations (same as e.g. STM32). Then board assigns N of these nodes to a `pinctrl-n` property. To my understanding, your pre-definitions are, e.g. `alt1_no_lpc_espi`, `lpc_espi_default` is a pre-populated choice. Am I missing something?

Reply:
Got your point. Preparing all valid pre-defined nodes in ```pinctrl``` node doesn't violate the principle. Then users can overwrite and assign them to a ```pinctrl-n``` property in the board DT file. Am I right? If so, I will prepare the other PR for this comment.

Reply:
Hi @gmarull, the new PR provides the pre-defined nodes with all valid combinations in ```pinctrl``` node only. Users still need to select these pre-defined nodes in ```pinctrl-0/1/2``` property of the board DT file. Since all child nodes of ```pinctrl``` node is a single mapping between pin muxing and its control register/bit, this PR doesn't use groups anymore.  Any comment is appreciated.